src/EA.Weee.RequestHandlers.Tests.Unit/Admin/Obligations/SubmitSchemeObligationsHandlerTests.cs

```diff
@@ -56,5 +140,74 @@ public async Task HandleAsync_GivenRequest_CsvDataShouldBeValidated()
             //assert
             A.CallTo(() => obligationUploadValidator.Validate(A<List<ObligationCsvUpload>>.That.Matches(o => o.SequenceEqual(obligationUploadData)))).MustHaveHappenedOnceExactly();
         }
+
+        [Fact]
+        public async Task HandleAsync_GivenRequestWithNoErrors_ObligationDataAccessShouldBeCalled()
+        {
+            //arrange
+            var obligationErrors = new List<ObligationUploadError>();
+            var authority = fixture.Create<UKCompetentAuthority>();
+            A.CallTo(() => obligationUploadValidator.Validate(A<List<ObligationCsvUpload>>._))
+                .Returns(obligationErrors);
+            A.CallTo(() => commonDataAccess.FetchCompetentAuthority(A<CompetentAuthority>._)).Returns(authority);
+
+            //act
+            await handler.HandleAsync(request);
+
+            //assert
+            A.CallTo(() => obligationDataAccess.AddObligationUpload(A<UKCompetentAuthority>.That.Matches(a => a.Equals(authority)),
+                A<string>.That.Matches(s => s.Equals(System.Text.Encoding.UTF8.GetString(request.FileInfo.Data))),
+                A<string>.That.Matches(s => s.Equals(request.FileInfo.FileName)), A<List<ObligationUploadError>>.That.Matches(o => o.SequenceEqual(obligationErrors)))).MustHaveHappenedOnceExactly();
+        }
+
+        [Fact]
+        public async Task HandleAsync_GivenRequestWithErrors_ObligationDataAccessShouldBeCalled()
+        {
+            //arrange
+            var obligationErrors = fixture.CreateMany<ObligationUploadError>().ToList();
+            var authority = fixture.Create<UKCompetentAuthority>();
+            
+            A.CallTo(() => obligationUploadValidator.Validate(A<List<ObligationCsvUpload>>._))
+                .Returns(obligationErrors);
+            A.CallTo(() => commonDataAccess.FetchCompetentAuthority(A<CompetentAuthority>._)).Returns(authority);
+
+            //act
+            await handler.HandleAsync(request);
+
+            //assert
+            A.CallTo(() => obligationDataAccess.AddObligationUpload(A<UKCompetentAuthority>.That.Matches(a => a.Equals(authority)),
```

Comment:
is the csv functionality only working for the UK Authority?

Reply:
UKCompetentAuthority is the domain object that makes to the Lookup.CompetentAuthority table in the database